browser/ui/views/brave_actions/brave_shields_action_view.cc

```diff
@@ -71,7 +72,16 @@ BraveShieldsActionView::BraveShieldsActionView(Profile* profile,
   tab_strip_model_->AddObserver(this);
 }
 
-BraveShieldsActionView::~BraveShieldsActionView() = default;
+BraveShieldsActionView::~BraveShieldsActionView() {
+  // The lifetime of this class can be shorter than the WebContents,
+  // and if that's true, the last registered web_contents won't unregister
+  // itself. By keeping a cache of last active web contents we ensure to
+  // unregister from this edge case.
+  if (active_web_contents_) {
```

Comment:
on second thought I think this should be `DCHECK(!active_web_contents_)` since a toolbar button should outlive the webcontents in the tab strip